input/posts/2020-09-02-migrating-wcf-to-grpc.md

```diff
@@ -0,0 +1,507 @@
+
+<pre><code class="language-csharp">
+[ServiceContract]
+public interface IPortfolioService
+{
+    [OperationContract]
+    Task&lt;Portfolio&gt; Get(<span style="background-color: #FFFF00"><b>GetPortfolioRequest request</b></span>);
+
+    [OperationContract]
+    Task&lt;<span style="background-color: #FFFF00"><b>PortfolioCollection</b></span>&gt; GetAll(<span style="background-color: #FFFF00"><b>GetAllPortfoliosRequest request</b></span>);
+}
+</code></pre>
+
+The above changes in the service contract forces us to create some additional data contracts. So we create the following:
+```csharp
+[DataContract]
+public class GetPortfolioRequest
+{
+    [DataMember(Order = 1)]
+    public Guid TraderId { get; set; }
+
+    [DataMember(Order = 2)]
+    public int PortfolioId { get; set; }
+}
+
+[DataContract]
+public class GetAllPortfoliosRequest
+{
+    [DataMember(Order = 1)]
+    public Guid TraderId { get; set; }
+}
+
+[DataContract]
+public class PortfolioCollection
+{
+    [DataMember(Order = 1)]
+    public List<Portfolio> Items { get; set; }
+}
+```
+
+That's basically it. Now we have migrated our WCF service contract and data contracts to gRPC. Next step is to migrate the data layer to .NET Core.
+
+## Migrate the PortfolioData library to .NET Core 
+
+Next up, we'll migrate the PortfolioData library to .NET Core, as described in Microsoft's guide [here](https://docs.microsoft.com/en-us/dotnet/architecture/grpc-for-wcf-developers/migrate-request-reply#migrate-the-portfoliodata-library-to-net-core). We will however not copy over the models (`Portfolio.cs` and `PortfolioItem.cs`) because these are already defined in the class library we created in the previous chapter. Instead we'll add a project reference to that shared library. Next step is to migrate the WCF service to an ASP.NET Core Application.
+
+## Migrate the WCF Service to an ASP.NET Core application
+
+The first thing we need to do is to create an ASP.NET Core application. So either start your favorite IDE and create a basic ASP.NET Core application or just simply run `dotnet new web` from the command-line. Next, we need to add a reference to the [protobuf-net.Grpc.AspNetCore](https://www.nuget.org/packages/protobuf-net.Grpc.AspNetCore/) NuGet package. Install it using you favorite package manager, or simply run `dotnet add package protobuf-net.Grpc.AspNetCore`. We'll also need to add a project reference to the PortfolioData library we created in the previous chapter.
```

Comment:
Consider removing "just simply" - I've been trying to take both words out of my blogging/Tweeting vocabulary